.changeset/fix-type-import-paths.md

```diff
@@ -0,0 +1,47 @@
+---
+"@devopness/ui-react": patch
+---
+
+fixes component exported types, unresolved paths aliases
+
+[tsconfig.json](./tsconfig.json) was configured to resolve imports using [rootDirs](https://www.typescriptlang.org/tsconfig/#rootDirs), which caused to import paths to not be resolved in build files.
+
+Solution was to declare root folder "src" as one of the [paths](https://www.typescriptlang.org/tsconfig/#paths) to be resolved.
+
+This patch should not require any changes to your code, if all props are correctly typed.
+
+BEFORE:
+
+[@devopness/ui-react@2.146.0 - npm Code tab](https://www.npmjs.com/package/@devopness/ui-react/v/2.146.0?activeTab=code)
+
+```ts
+// dist/components/Buttons/Button/Button.d.ts#L2-L3
+import { getColor } from 'src/colors';
+import { Icon } from 'src/icons';
+```
+
+This resulted in props like `ButtonProps.icon`  being `any`
+
+```ts
+import { Button } from '@devopness/ui-react'
+
+// ...
+
+<Button icon="non-existing-icon" /> // => no errors, accepts any value
+```
+
+AFTER:
+
+```ts
+// dist/components/Buttons/Button/Button.d.ts#L2-L3
+import { getColor } from '../../../colors';
+import { Icon } from '../../../icons';
```

Comment:
Please explain in code why `AFTER` is recommended instead of the `BEFORE` approach.

It seems not obvious to understand the advantage of importing using relative paths `../.../...` whereas [tsconfig.json defines `src` path](https://github.com/devopness/devopness/pull/1007/files#diff-b29ce0c0c2c42d45077f7ec5f6a4a8ae9091014a4cd164bc435cefd0226e4848R14).

Reply:
> It seems not obvious to understand the advantage of importing using relative paths ../.../... whereas tsconfig.json defines src path.

@jfoliveira I believe there is some misunderstanding, I'm not proposing a change in the source code files `src/*`, I'm solving a bug in build files `dist/` published to npm.

![image](https://github.com/user-attachments/assets/6f3fd8f5-82f7-4326-888d-f3998bf10ff9)

The current version published to npm does not resolve types imported from "src", "src" folder does not exist after build.

With that in mind, could you give me feedback on how to improve the changelog? 

Currently, I've added the following to help highlight this a build related change

> fixes component **exported** types, unresolved paths aliases

> which caused to import paths to not be resolved in **build** files.

> This resulted in props like `ButtonProps.icon` being **`any`**

> // **dist**/components/Buttons/Button/Button.d.ts#L2-L3

Reply:
> I'm not proposing a change in the source code files src/*, I'm solving a bug in build files dist/ published to npm.

Got it!
All good then :-)